pkg/cloudprovider/provider/vsphere/helper.go

```diff
@@ -236,6 +236,12 @@ func generateLocalUserdataIso(userdata, name string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to create local temp directory for userdata at %s: %v", userdataDir, err)
 	}
+	defer func(dir string) {
+		err := os.RemoveAll(userdataDir)
+		if err != nil {
+			glog.Errorf("error cleaning up local userdata tempdir %s: %v", userdataDir, err)
```

Comment:
You have a parameter for your defer func but still use the `userdataDir` variable from the parent scope.
Just relying on the parent scope is fine here. No need to specify a parameter